### PR TITLE
Contentful/export locales

### PR DESCRIPTION
**Pull request by dpinol**
## Description

Tool to modify contentful spaces exported as json file with "contentful space export".

## Context

Some times we want to clone the contents of a space into another one but changing a locale (but keeping the field values) and removing some locales (important because the limitation in number of locales based on the space type) 

## Approach taken / Explain the design

Existing tools to import/export csv file for translators have been joined with the new tool in a l10n (localization) folder. 

## To document / Usage example

Execute "bin/l10n/migrate-locales.sh --help" to get instructions

## Testing

The pull request...

- has unit tests

### CODE REVIEW

File: packages/botonic-plugin-contentful/src/contentful/export/locale-migrator.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  defaultLoc: string | undefined
+  /**
+   * @param fromLoc The locale whose flow we want to maintain
+   * @param toLoc The locale in which `fromLoc`'s flow will be converted
+   */
+  constructor(
+    readonly fromLoc: string,
+    readonly toLoc: string,
+    readonly verbose = false
+  ) {}
```
Comment (asastre) [inline packages/botonic-plugin-contentful/src/contentful/export/locale-migrator.ts:19]: The parameter assignment to class attributes is missing. How is this working?

File: packages/botonic-plugin-contentful/src/contentful/export/locale-migrator.ts [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      spaceExport.payload.locales = locales
+    }
+  }
+}
+
+export class LocaleRemover {
+  /**
+   * @param removeLocs The locales to completely remove
+   */
+  constructor(readonly removeLocs: string[], readonly newDefault?: string) {}
```
Comment (asastre) [inline packages/botonic-plugin-contentful/src/contentful/export/locale-migrator.ts:140]: Same here, there are no assignments to class attributes. How is this working?

File: packages/botonic-plugin-contentful/src/tools/l10n/locale-migrate.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import {
+  LocaleMigrator,
+  LocaleRemover,
+} from '../../contentful/export/locale-migrator'
+import { SpaceExport } from '../../contentful/export/space-export'
+
+if (process.argv.length != 7 || process.argv[2] == '--help') {
+  console.error(
+    `Usage: fromFile toFile fromLocale locale toLocale [removeLocales]`
```
[the remark this answers is not in the document]
Comment (dpinol) [reply packages/botonic-plugin-contentful/src/tools/l10n/locale-migrate.ts]: fixed, thanks

File: packages/botonic-plugin-contentful/src/tools/l10n/locale-migrate.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import {
+  LocaleMigrator,
+  LocaleRemover,
+} from '../../contentful/export/locale-migrator'
+import { SpaceExport } from '../../contentful/export/space-export'
+
+if (process.argv.length != 7 || process.argv[2] == '--help') {
```
[the remark this answers is not in the document]
Comment (dpinol) [reply packages/botonic-plugin-contentful/src/tools/l10n/locale-migrate.ts]: also true, thanks! :+1:

File: packages/botonic-plugin-contentful/src/contentful/export/locale-migrator.ts (hunk 1 above)
Comment (dpinol) [reply packages/botonic-plugin-contentful/src/contentful/export/locale-migrator.ts:19]: this is typescript! just having readonly, public or private does the declaration of the parameter, class attribute and assignment in one go :-)

File: packages/botonic-plugin-contentful/src/tools/l10n/locale-migrate.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+const toFile = process.argv[3]
+const fromLocale = process.argv[4]
+const toLocale = process.argv[5]
+const removeLocales = process.argv.length > 6 ? process.argv[6] : ''
+
+function main() {
+  try {
+    const spaceExport = SpaceExport.fromJsonFile(fromFile)
+    const migrator = new LocaleMigrator(fromLocale, toLocale)
+    const remover = new LocaleRemover(removeLocales.split(','), toLocale)
```
[the remark this answers is not in the document]
Comment (dpinol) [reply packages/botonic-plugin-contentful/src/tools/l10n/locale-migrate.ts:23]: cool, added to #896

**Pull request merged**